iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -565,11 +568,20 @@ private TransportState MoveToStateIfPossible(TransportState destination, Transpo
 
         private async Task OpenAsyncInternal(CancellationToken cancellationToken)
         {
+            if (_webProxy != DefaultWebProxySettings.Instance)
```

Comment:
Might put this logic in a helper method, because it is not necessarily an obvious conclusion by someone new to the code, and we repeat it several times.

Reply:
Sure thing